### PR TITLE
Style mirror

### PR DESCRIPTION
**Pull request by aaroncox**

### CODE REVIEW

File: src/account.ts [hunk 1]
```diff
@@ -4,154 +4,198 @@ import {
...
+            receiver = Name.from(options.receiver)
+        }
+        return this.systemContract.action('buyram', {
+            payer: this.accountName,
+            quant: amount,
+            receiver,
         })
     }
 
-    getResources(): Resources {
```
Comment (dafuga) [inline src/account.ts:118]: Don't you think that there is value in having a method that returns all resource data?

File: src/permission.ts [hunk 2]
```diff
@@ -36,133 +37,123 @@ export interface ActionData {
...
-                    weight: Number(weight),
-                })),
-                waits: this.permission_data.auth?.waits?.map(({wait_sec, weight}) => ({
-                    wait_sec: Number(wait_sec),
-                    weight: Number(weight),
-                })),
-                threshold: Number(this.permission_data.auth?.threshold),
-            }),
+@Struct.type('permission')
+export class Permission extends Struct {
```
Comment (dafuga) [inline src/permission.ts:61]: This is so much better! Not sure why I didn't think of making Permission a Struct 😅

File: src/account.ts [hunk 3]
```diff
@@ -4,154 +4,198 @@ import {
...
+        return new Resource(resourceType, this.data)
     }
 
-    removePermission(permissionName: NameType): Action {
-        return this.eosioContract.action('deleteauth', {
-            account: '............1',
-            authorized_by: '............1',
-            permission: permissionName,
+    // TODO: Refactor once resources library is updated
+    resources(sampleAccount?: NameType) {
```
Comment (dafuga) [inline src/account.ts:114]: What is the `sampleAccount` here? 🤔 Should this be the current account?

File: src/resource.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+export type ResourceType = 'cpu' | 'net' | 'ram'
+
+export interface ResourceValues {
+    available: Int64
+    used: Int64
+    max: Int64
+}
+
+export class Resource {
```
Comment (dafuga) [inline src/resource.ts:11]: 💯

**Review by dafuga — APPROVED** [review]
Looks great!

File: src/account.ts (hunk 3 above)
Comment (aaroncox) [reply src/account.ts:114]: It's a frequently used account on the network, where the resource values are constantly refreshing due to usage. For that reason it's not great to have it be the users account. 

Kinda a hacky way to calculate this stuff, and an optional value for the resources library. I'm hoping most people won't have to use it or we'll find a way to get rid of it all together.

**Pull request merged**